src/containers/AuthenticationPage/AuthenticationPage.js

```diff
@@ -136,10 +172,88 @@ export class AuthenticationPageComponent extends Component {
       submitSignup(params);
     };
 
-    const formContent = (
+    const handleSubmitConfirm = values => {
+      const { idpToken, email, firstName, lastName, idpId } = this.state.authInfo;
+      const { email: newEmail, firstName: newFirstName, lastName: newLastName } = values;
+
+      // Pass email, fistName or lastName to Flex API only if user has edited them
+      // sand they can't be fetched directly from idp provider (e.g. Facebook)
+
+      const authParams = {
+        ...(newEmail !== email && { email: newEmail }),
+        ...(newFirstName !== firstName && { firstName: newFirstName }),
+        ...(newLastName !== lastName && { lastName: newLastName }),
+      };
+
+      submitSinguoWithIdp({
+        idpToken,
+        idpId,
+        ...authParams,
+      });
+    };
+
+    const authWithFacebook = () => {
+      const baseUrl = apiBaseUrl();
+
+      if (from) {
+        window.location.href = `${baseUrl}/api/auth/facebook?from=${from}`;
```

Comment:
I really think we should include other possible return paths as URL params too!

That way we don't need to hard-code web app's internal routes all over the place.
I think **landing page** and **confirm page** are possible paths where the user might end up.

It would also be good to add some code comment to here. What is this "from"? what is it doing?

Reply:
You could also lose the if-else pattern if there more params. You could create something like this:

```js
const fromParam = from ? `from=${from}` : '';
const defaultReturnParam = defaultReturn ? `&defaultReturn=${defaultReturn}` : '';
window.location.href = `${baseUrl}/api/auth/facebook?${fromParam}${defaultReturnParam}`;
```

BUT we do have [query-string](https://www.npmjs.com/package/query-string#stringifyurlobject-options) package as a dependency.